### PR TITLE
Make --no-build imply --no-restore for run, pack, and test commands.

### PR DESCRIPTION
**Pull request by peterhuene**
These commits makes the `--no-build` option for the `run`, `pack`, and `test` commands automatically imply `--no-restore`.

Additionally, the verbosity option has been added to the `run` command to support tests.

Fixes issue #7472.

### CODE REVIEW

**Comment by livarcocc** [comment]
@nguerrera can you please review this?

@peterhuene do you mind checking dotnet run and dotnet pack? Both of these commands depend on build and implicit restore and I think it makes sense to check that this is fixed across all command.

To be complete, build and publish also do implicit restore, but these two you don't have --no-build as an option.

**Comment by peterhuene** [comment]
I'll push up fixes for `run` and `pack` as well.

**Comment by nguerrera** [comment]
Looks good to me so far. :) I'll review again when the other commands are fixed.

**Comment by peterhuene** [comment]
@nguerrera I've pushed up a few more commits for the other two commands.  The one question I have is about adding the verbosity option to the run command; this was necessary to see what the run command was doing for testing, but I think it's useful on its own.

Re: the removing of the colon commit: is it better to remove it from the string resource or the format string?  I opted for the latter, but perhaps it should be done in the former.

**Comment by nguerrera** [comment]
> The one question I have is about adding the verbosity option to the run command; this was necessary to see what the run command was doing for testing, but I think it's useful on its own.

This is a nice change. Let's keep it. :)

> Re: the removing of the colon commit: is it better to remove it from the string resource or the format string? I opted for the latter, but perhaps it should be done in the former.

I think it's fine the way you've done it and it has the advantage of not needing a translation cycle to be fixed in other languages.

File: src/dotnet/commands/dotnet-run/RunCommandParser.cs [hunk 1]
```diff
@@ -51,7 +51,8 @@ internal static class RunCommandParser
                         "--no-build",
                         LocalizableStrings.CommandOptionNoBuildDescription,
                         Accept.NoArguments()),
-                    CommonOptions.NoRestoreOption()
+                    CommonOptions.NoRestoreOption(),
+                    CommonOptions.VerbosityOption()
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-run/RunCommandParser.cs]: There is a bug out for setting the verbosity for run. I will find and reference it here. We can use this PR to close both issues.

File: test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs [hunk 2]
```diff
@@ -75,6 +75,25 @@ public void ItCanRunAMultiTFMProjectWithImplicitRestore()
                          .And.HaveStdOutContaining("This string came from the test library!");
         }
 
+        [Fact]
```
Comment (livarcocc) [inline test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs]: Should we add a test for the verbosity option?

File: src/dotnet/commands/dotnet-run/RunCommandParser.cs (hunk 1 above)
Comment (peterhuene) [reply src/dotnet/commands/dotnet-run/RunCommandParser.cs]: I'll amend the commit message with a reference.

File: test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs (hunk 2 above)
Comment (peterhuene) [reply test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs]: I'll amend the commit with one.

File: src/dotnet/commands/dotnet-run/RunCommandParser.cs (hunk 1 above)
Comment (peterhuene) [reply src/dotnet/commands/dotnet-run/RunCommandParser.cs]: Is it #4823 (partial fix here) or another issue?

File: test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs (hunk 2 above)
Comment (peterhuene) [reply test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs]: Added a test to that commit to check that the verbosity option does something.  Waiting on an issue to reference, if there is an existing one, before pushing it up.

File: src/dotnet/commands/dotnet-run/RunCommandParser.cs (hunk 1 above)
Comment (livarcocc) [reply src/dotnet/commands/dotnet-run/RunCommandParser.cs]: https://github.com/dotnet/cli/issues/7932

File: test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs (hunk 2 above)
Comment (peterhuene) [reply test/dotnet-run.Tests/GivenDotnetRunRunsCsProj.cs]: Pushed in ea85555.

File: src/dotnet/commands/dotnet-run/RunCommandParser.cs (hunk 1 above)
Comment (peterhuene) [reply src/dotnet/commands/dotnet-run/RunCommandParser.cs]: Pushed in ea85555.

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x64 Release Build

**Comment by livarcocc** [comment]
Ubuntu16.04 has actually succeeded, but for some reason the PR did not get updated. cc @mmitche

**Comment by mmitche** [comment]
@livarcocc No idea, nothing in the log shows an obvious error.  The update at the end is just missing.

**Pull request merged**